tests/gemv.cpp

```diff
@@ -0,0 +1,164 @@
+          A(i,j) = (double(i)+startVal) +
+            (double(j)+startVal) * double(A_numRows);
+        }
+      }
+    }
+  };
+
+  template<class MdspanType, class Scalar>
+  void fill_matrix(MdspanType A, const Scalar startVal) {
+    FillMatrix<MdspanType, Scalar>::fill(A, startVal);
+  }
+
+  template<class MdspanType, class Scalar>
+  struct FillVector {
+    static void fill(MdspanType x, const Scalar startVal);
+  };
+
+  template<class MdspanType>
+  struct FillVector<MdspanType, int> {
+    static void fill(MdspanType x, const int startVal)
+    {
+      const ptrdiff_t numRows = x.extent(0);
+      for (ptrdiff_t i = 0; i < numRows; ++i) {
+        x(i) = int(i+startVal);
+      }
+    }
+  };
+
+  template<class MdspanType>
+  struct FillVector<MdspanType, double> {
+    static void fill(MdspanType x, const double startVal)
+    {
+      const ptrdiff_t numRows = x.extent(0);
+      for (ptrdiff_t i = 0; i < numRows; ++i) {
+        x(i) = double(i+startVal);
+      }
+    }
+  };
+
+  template<class MdspanType, class Scalar>
+  void fill_vector(MdspanType x, const Scalar startVal) {
+    FillVector<MdspanType, Scalar>::fill(x, startVal);
+  }
+
+  template<class Scalar>
+  struct Magnitude {
+    using type = Scalar;
+  };
+  template<class Real>
+  struct Magnitude<std::complex<Real>> {
+    using type = Real;
+  };
+
+  template<class Scalar>
+  void test_matrix_product()
+  {
+    using scalar_t = Scalar;
+    using real_t = typename Magnitude<Scalar>::type;
+
+    using extents_t = extents<dynamic_extent, dynamic_extent>;
+    using matrix_t = basic_mdspan<scalar_t, extents_t, layout_left>;
+    using vector_t = basic_mdspan<scalar_t, extents<dynamic_extent>, layout_left>;
+
+    constexpr ptrdiff_t maxDim = 7;
+    constexpr ptrdiff_t storageSize(maxDim*maxDim + 3*maxDim);
+    std::vector<scalar_t> storage(storageSize);
+
+    for (ptrdiff_t numRows : {1, 4, 7}) {
+      for (ptrdiff_t numCols : {1, 4, 7}) {
+        ptrdiff_t offset = 0;
+        matrix_t A(storage.data() + offset, numRows, numCols);
+        offset += numRows * numCols;
+        vector_t x(storage.data() + offset, numCols);
+        offset += numCols;
+        vector_t y(storage.data() + offset, numRows);
+        offset += numRows;
+        vector_t gs(storage.data() + offset, numRows);
+
+        fill_matrix(A, scalar_t(real_t(1)));
+        fill_vector(x, scalar_t(real_t(2)));
+
+        // Initialize vector to zero
+        for (ptrdiff_t i = 0; i < numRows; i++) {
+          gs(i) = scalar_t(0.0); // this works even for complex
+        }
+
+        // Perform matvec
+        for (ptrdiff_t j = 0; j < numCols; ++j) {
+          for (ptrdiff_t i = 0; i < numRows; ++i) {
+            gs(i) += A(i,j) * x(j);
+          }
+        }
+
+        // Fill result vector with flag values to make sure that we
+        // computed everything.
+        for (ptrdiff_t j = 0; j < numRows; ++j) {
+          y(j) = std::numeric_limits<scalar_t>::min();
```

Comment:
For floating-point types, `min()` actually returns the smallest normalized positive value:

https://en.cppreference.com/w/cpp/types/numeric_limits/min

That's probably fine for this test -- it's just likely not what you meant.